src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -120,9 +170,20 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 		if (!result) {
 			return;
 		}
-		const { fixes, onDidRunQuickFix } = result;
+		const { fixes, onDidRunQuickFix, expectedCommands } = result;
+		this._expectedCommands = expectedCommands;
 		this._quickFixes = fixes;
-		onDidRunQuickFix(() => this._disposeQuickFix());
+		this._register(this.onDidRequestRerunCommand(() => {
+			this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixExecuted);
+			this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixExecuted);
+			this._fixesAvailable = false;
+		}));
+		this._register(onDidRunQuickFix(() => {
+			this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixExecuted);
+			this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixExecuted);
```

Comment:
Won't this double report? I forgot what `onDidRequestRerunCommand` is for

Reply:
doesn't double report based on my testing. `onDidRunQuickFix` only gets fired for quick fixes without a type or of type `open`

Reply:
It does, it was being sent 3 times when I ran a quick fix:

![image](https://user-images.githubusercontent.com/2193314/195664053-d2e0ac64-cd67-4a53-afc4-4cc2cc1fa229.png)

Reply:
It might be a listener not being disposed, after a bunch of testing it's reporting it a lot more:

![image](https://user-images.githubusercontent.com/2193314/195665087-e7f52f10-a900-4dde-b5e4-46718465ec7e.png)